zyte_common_items/adapter.py

```diff
@@ -99,3 +99,22 @@ def __iter__(self) -> Iterator:
             if not _is_empty(self.item._unknown_fields_dict[attr])
         )
         return iter(fields)
+
+
+class ZyteKeepEmptyAdapter(ZyteItemAdapter):
+    """Similar to :class:`~.ZyteItemAdapter` but doesn't remove empty values.
+
+    It is intended to be used in tests and other use cases where it's important
+    to differentiate between empty and missing fields.
+    """
+
+    def __iter__(self) -> Iterator:
+        fields = [attr for attr in self._fields_dict if hasattr(self.item, attr)]
+        fields.extend(self.item._unknown_fields_dict)
+        return iter(fields)
+
+
+class ZyteKeepEmptyItemAdapter(ItemAdapter):
```

Comment:
I've added this and tests for it but haven't documented it yet. Initially it was created for easier test fixture creation but as it's now general purpose it looks like we either want `ItemAdapter` subclasses for both `ZyteItemAdapter` and `ZyteKeepEmptyAdapter` or for none of them, not just for the second one.

Also there is a serious naming problem here, mostly for an `ItemAdapter` subclass that enables `ZyteItemAdapter`.

Reply:
I've implemented most of this, should we also now rename ` ZyteItemAdapter` to `ZyteItemNoEmptyAdapter` and deprecate the old name?

Reply:
+0.2 from me to keep the ZyteItemAdapter name, to emphasize that's the recommended approach.